Docker/Runner/DataLoader/DataLoader.php

```diff
@@ -180,53 +190,46 @@ public function storeOutput()
     }
 
     /**
-     * Archive data directory and save it to Storage, do not actually run the container.
+     * Archive data directory and save it to Storage
      * @param array $tags Arbitrary storage tags
+     * @throws ClientException
      */
     public function storeDataArchive(array $tags)
     {
         $zip = new \ZipArchive();
-        $zipFileName = 'data.zip';
-        $zipDir = $this->dataDirectory . DIRECTORY_SEPARATOR . '..' . DIRECTORY_SEPARATOR . 'zip';
-        $fs = new Filesystem();
-        $fs->mkdir($zipDir);
-        $zip->open($zipDir. DIRECTORY_SEPARATOR . $zipFileName, \ZipArchive::CREATE);
+        $zipFileName = $this->dataDirectory . DIRECTORY_SEPARATOR . 'data.zip';
+        $zip->open($zipFileName, \ZipArchive::CREATE);
         $finder = new Finder();
         /** @var SplFileInfo $item */
         foreach ($finder->in($this->dataDirectory) as $item) {
-            if ($item->getPathname() == $zipDir) {
-                continue;
-            }
             if ($item->isDir()) {
                 if (!$zip->addEmptyDir($item->getRelativePathname())) {
-                    throw new ApplicationException("Failed to add directory: ".$item->getFilename());
+                    throw new ApplicationException("Failed to add directory: " . $item->getFilename());
                 }
             } else {
-                if (!$zip->addFile($item->getPathname(), $item->getRelativePathname())) {
-                    throw new ApplicationException("Failed to add file: ".$item->getFilename());
+                if ($item->getPathname() == $zipFileName) {
+                    continue;
+                }
+                if (($item->getRelativePathname() == 'config.json') || ($item->getRelativePathname() == 'state.json')) {
```

Comment:
Takže tohle teoreticky může vzít `state.json` když ho vygeneruje i procesor? Nemělo by se to chovat stejně jako tady?

https://github.com/keboola/docker-bundle/blob/d0b76dd77693c3d24e1a8a6d5ec477eb842074fc/Service/Runner.php#L437